javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +518,971 @@ object Character {
+      map.put("wancho", WANCHO)
+      map.put("mendekikakui", MENDE_KIKAKUI)
+      map.put("adlam", ADLAM)
+      map.put("indicsiyaqnumbers", INDIC_SIYAQ_NUMBERS)
+      map.put("ottomansiyaqnumbers", OTTOMAN_SIYAQ_NUMBERS)
+      map.put("arabicmathematicalalphabeticsymbols", ARABIC_MATHEMATICAL_ALPHABETIC_SYMBOLS)
+      map.put("mahjongtiles", MAHJONG_TILES)
+      map.put("dominotiles", DOMINO_TILES)
+      map.put("playingcards", PLAYING_CARDS)
+      map.put("enclosedalphanumericsupplement", ENCLOSED_ALPHANUMERIC_SUPPLEMENT)
+      map.put("enclosedideographicsupplement", ENCLOSED_IDEOGRAPHIC_SUPPLEMENT)
+      map.put("emoticons", EMOTICONS)
+      map.put("ornamentaldingbats", ORNAMENTAL_DINGBATS)
+      map.put("alchemicalsymbols", ANATOLIAN_HIEROGLYPHS)
+      map.put("geometricshapesextended", GEOMETRIC_SHAPES_EXTENDED)
+      map.put("supplementalarrowsc", SUPPLEMENTAL_ARROWS_C)
+      map.put("chesssymbols", CHESS_SYMBOLS)
+      map.put("cjkunifiedideographsextensionb", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B)
+      map.put("cjkunifiedideographsextensionc", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C)
+      map.put("cjkunifiedideographsextensiond", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D)
+      map.put("cjkunifiedideographsextensione", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_E)
+      map.put("cjkunifiedideographsextensionf", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_F)
+      map.put("cjkcompatibilityideographssupplement", CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT)
+      map.put("tags", TAGS)
+      map.put("variationselectorssupplement", VARIATION_SELECTORS_SUPPLEMENT)
+      map.put("supplementaryprivateuseareaa", SUPPLEMENTARY_PRIVATE_USE_AREA_A)
+      map.put("supplementaryprivateuseareab", SUPPLEMENTARY_PRIVATE_USE_AREA_B)
+      map
+    }
+
+    // From: https://unicode.org/Public/UCD/latest/ucd/Blocks.txt
+    // Note:   When comparing block names, casing, whitespace, hyphens,
+    //         and underbars are ignored.
+    //         For example, "Latin Extended-A" and "latin extended a" are equivalent.
+    //         For more information on the comparison of property values,
+    //           see UAX #44: http://www.unicode.org/reports/tr44/
+    private def toNormalizedName(name: String): String = {
+      name.toLowerCase.replaceAll(raw"[\s\-_]", "")
+    }
+
+    ////////////////
+    // End Generated
+    ////////////////
+
+    def forName(blockName: String): UnicodeBlock = {
+      val key: String = toNormalizedName(blockName)
+      if (!blocksByNormalizedName.containsKey(key)) throw new IllegalArgumentException()
+
+      blocksByNormalizedName.get(key)
+    }
+
+    def of(c: scala.Char): UnicodeBlock = of(c.toInt)
+
+    def of(codePoint: scala.Int): UnicodeBlock = {
+      // scalastyle:off return
+      if (!Character.isValidCodePoint(codePoint)) {
+        throw new IllegalArgumentException()
+      }
+
+      var low: Int  = 0
+      var mid: Int  = -1
+      var high: Int = allBlocks.length - 1
```

Comment:
In the Scala.js coding style, the only thing we ever vertically align are `=>` of consecutive one-liner `case`s.
```suggestion
      var low: Int = 0
      var mid: Int = -1
      var high: Int = allBlocks.length - 1
```